### ISSUE TITLE
SignColumn in Neogit

### ISSUE THREAD

**Issue by windowsrefund** [ramojus/mellifluous.nvim#74]
Greets,

Neogit makes use of the sign column to show the status of internal folds. 

![image](https://github.com/user-attachments/assets/02971d52-a546-4b20-bfb7-63ad72090b63)

![image](https://github.com/user-attachments/assets/92e860be-404f-4598-acf9-5e2df93b85bc)

When Mellifluous is active, `SignColumn` is linked to `LineNr`. Normally, this is great but we can see how it breaks down against the `NeogitStatus` file type.

By comparison, a look at the same thing via the default Astronvim theme has that area of the tool looking much cleaner.

![image](https://github.com/user-attachments/assets/4fa34ff1-c748-4377-96b2-a7dae2d0e3a3)

and I see when that theme is active, `SignColumn` is `cleared`.

So back to our favorite scheme, Mellifluous, I can't understand why that sign column is behaving like this? Even though it's linked to `LineNr`, it is behaving in a way where the bg color is highlighting when it should just sit there and remain dark. Of course, the real move here would be to simply ignore it and move on but I'm finding that to be a challenge :)

**Comment by ramojus** [comment]
Yeah, it doesn't look good, the backgrounds in signcolumn should match. I don't use Neogit, so I'm not really motivated to investigate this. There might be some Neogit highlight group that you could set, or you can clear the `SignColumn` highlight group.

**Comment by windowsrefund** [comment]
Turns out things are good when `foldcolumn` is used rather than `signcolumn`. I've asked the hard-coded options be removed [here](https://github.com/NeogitOrg/neogit/issues/1536)

**Comment by ramojus** [comment]
That is strange
![image](https://github.com/user-attachments/assets/234528d9-dd2a-44d8-86c4-8fe6ad7496c5)
![image](https://github.com/user-attachments/assets/2733f918-96fc-4637-a305-f68e56fc8cc7)

So it respects the background of foldcolumn, but doesn't respect the background of signcolumn? Or do you use different `FoldColumn` highlight than default?

**Comment by windowsrefund** [comment]
I was saying that everything works as expected when `foldcolumn` is enabled and `signcolumn` is not. My understanding is that Neogit essentially displays the same information in both and that the developer has decided to hard-code setting `foldcolumn=0`. Seeing if I can convince them to not hard-code anything so the end user can decide where they want the folds displayed. As to why the SignColumn highlighting isn't working, I don't know but it won't matter if I can just lean into using `foldcolumn`.
